common/src/unifyfs_const.h

```diff
@@ -49,6 +49,9 @@
 #define UNIFYFS_MAX_FILENAME KIB
 #define UNIFYFS_MAX_HOSTNAME 64
 
+/* key + '/' value string */
```

Comment:
@tonyhutter I understand the intention of why you added this, but my assertion is that no file or directory path should ever be greater than UNIFYFS_MAX_FILENAME. If we don't think the current value (1 KiB) is big enough, we should increase its size, rather than defining a new constant for one particular use case.

Reply:
The problem is that we do a bunch of `sprintf(buf, %s/%s", str1, str2)` where `buf`, `str1` and `str2` are each UNIFYFS_MAX_FILENAME in length, so increasing UNIFYFS_MAX_FILENAME won't help get rid of the compiler warning.

Reply:
This is the reason I always use `snprintf()` rather than `sprintf()`.

Reply:
There are ways to pacify the compiler for the keyval code, but they are less than ideal from a code cleanliness point of view. One option would be to define UNIFYFS_MAX_DIR_NAMELEN (max name length of a directory path, including any trailing slash) and UNIFYFS_MAX_DIRENTRY_NAMELEN (max name length of a directory entry) and then make UNIFYFS_MAX_FILENAME = UNIFYFS_MAX_DIR_NAMELEN + UNIFYFS_MAX_DIRENTRY_NAMELEN. To me, that's ugly, but it would make the compiler happy.

Reply:
snprintf() doesn't help.   It complains that we should be checking the return code since we could be truncating the buffer.
```C
        char str1[1024] = "a";
        char str2[1024] = "b";
        char buf[1024];

        snprintf(buf, 1024, "%s/%s", str1, str2);
```
```
error: ‘%s’ directive output may be truncated writing up to 1023 bytes into a region of size between 0 and 1023 [-Werror=format-truncation=]
  snprintf(buf, 1024, "%s/%s", str1, str2);
```
Thats why I added `UNIFYFS_MAX_KVFILENAME` to get around it ever truncating.

Reply:
This would avoid the compiler warning, but it seems excessive to me.
```
#define UNIFYFS_MAX_DIRENTRY_NAMELEN 64
#define UNIFYFS_MAX_DIR_NAMELEN (UNIFYFS_MAX_FILENAME - UNIFYFS_MAX_DIRENTRY_NAMELEN)
char dirname[UNIFYFS_MAX_DIR_NAMELEN];
char entryname[UNIFYFS_MAX_DIRENTRY_NAMELEN];
char buf[UNIFYFS_MAX_FILENAME+2]; // +2 allows for '/' and NUL
snprintf(buf, sizeof(buf), "%s/%s", dirname, entryname);
```

Reply:
@MichaelBrim yea, no matter how it's done, it's going to be a pain to get all these buffers sized correctly.  Might be better to just implement the kernel's `scnprintf()` function:

https://lwn.net/Articles/69419/

```C
/*
 * This is a re-implementation of the Linux kernel's scnprintf() function.
 *
 * It's snprintf() but returns the number of chars actually written into buf[]
 * not including the '\0'.  It also avoids the -Wformat-truncation warnings.
 */
int scnprintf(char *buf, size_t size, const char *fmt, ...)
{
        va_list args;
        int rc;

        va_start(args, fmt);
        rc = vsnprintf(buf, size, fmt, args);
        va_end(args);

        if (rc >= size) {
                /* We truncated */
                return size - 1;
        }

        return rc;
}
```

Reply:
I ended up implementing `scnprintf()` in my latest push.  This gets rid of the need for all the buffer resizing.